datacube/storage/_read.py

```diff
@@ -188,7 +189,8 @@ def norm_read_args(roi, shape, extra_dim_index):
                         src_nodata=rdr.nodata, dst_nodata=dst_nodata)
         else:
             rio_reproject(pix, dst, src_gbox, dst_gbox, resampling,
-                          src_nodata=rdr.nodata, dst_nodata=dst_nodata)
+                          src_nodata=rdr.nodata, dst_nodata=dst_nodata,
+                          XSCALE=scale_xy[0], YSCALE=scale_xy[1])
```

Comment:
prefer use de-structure over indexing:

```suggestion
                          XSCALE=xscale, YSCALE=yscale)
```

main reason is that errors are easier to locate that way